src/main/java/com/exactpro/th2/crawler/messages/strategy/MessagesStrategy.java

```diff
@@ -148,22 +149,26 @@ public MessagesCrawlerData requestData(@NotNull Timestamp start, @NotNull Timest
                 .build();
         SearchResult<MessageData> result = CrawlerUtils.searchMessages(provider, searchParams, metrics);
         List<MessageData> messages = result.getData();
-        resumeIds = requireNonNull(result.getStreamsInfo(), "response from data provider does not have the StreamsInfo")
+
+        resumeIds = new HashMap<>(resumeIds == null ? Collections.emptyMap() : resumeIds);
+        Map<StreamKey, MessageID> nextResumeIds = requireNonNull(result.getStreamsInfo(), "response from data provider does not have the StreamsInfo")
                 .getStreamsList()
                 .stream()
                 .collect(toUnmodifiableMap(
                         stream -> new StreamKey(stream.getSession(), stream.getDirection()),
                         Stream::getLastId,
                         BinaryOperator.maxBy(Comparator.comparingLong(MessageID::getSequence))
                 ));
+
+        putAndCheck(nextResumeIds, resumeIds, "computing resume IDs");
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("New resume ids: {}", resumeIds.entrySet().stream()
```

Comment:
```suggestion
            LOGGER.debug("New resume ids: {}", nextResumeIds.entrySet().stream()
```

Reply:
We need to log the resulting collection of IDs. The `nextResumeIds` is only a response from provider and it is not yet the final result